api/internal/target/kusttarget.go

```diff
@@ -250,6 +258,37 @@ func (kt *KustTarget) accumulateTarget(ra *accumulator.ResAccumulator) (
 	return ra, nil
 }
 
+// TODO: add details on whether this is a trusted catalog or not.
+func LoadCatalogFile(ldr ifc.Loader, paths []string) ([]framework.Catalog, error) {
+	catalogs := []framework.Catalog{}
+
+	if len(paths) == 0 {
+		return catalogs, nil
+	}
+
+	for _, path := range paths {
+		content, err := ldr.Load(path)
+		if err != nil {
+			return nil, errors.WrapPrefixf(err, "unable to load catalog path")
+		}
+
+		j, err := yaml.YAMLToJSON(content)
+		if err != nil {
+			return nil, errors.WrapPrefixf(err, "invalid catalog")
+		}
+
+		dec := json.NewDecoder(bytes.NewReader(j))
+		dec.DisallowUnknownFields()
+
+		var c framework.Catalog
+		if err := dec.Decode(&c); err != nil {
+			return nil, errors.WrapPrefixf(err, "unable to find catalog")
+		}
+		catalogs = append(catalogs, c)
```

Comment:
Better to move this logic to a separate function for better readability, keeping the methods short and better testing.